uint8 width
uint8 height
uint16[] data
